File: 0326-power-of-three/solution.py
class Solution:
    def isPowerOfThree(self, n: int) -> bool:
        if n == 1:
            return True
        while n != 0:
            n /= 3
            if math.floor(n) != n:
                return False
            if n == 1:
                return True
        return False
        

File: 0326-power-of-three/solution.cpp
class Solution {
public:
    bool isPowerOfThree(int n) 
    {
        if(n == 0) return false;
        
        if(n == 1 || n == 3)
        {
            return true;
        }
        
        while(n%3 == 0)
        {
            n /= 3;
        }
        
        return n == 1;
        
    }
};
